swupd/create_manifests.go

```diff
@@ -322,7 +332,7 @@ func aggregateManifests(newManifests []*Manifest, newMoM *Manifest, version uint
 }
 
 // CreateManifests creates update manifests for changed and added bundles for <version>
-func CreateManifests(version uint32, minVersion uint32, format uint, statedir string) (*MoM, error) {
+func CreateManifests(version uint32, minVersion uint32, format uint, statedir string, numWorkers int) (*MoM, error) {
```

Comment:
Ok, you mentioned that numWorkers will never be 0, yet it could very easily be set to 0 by the caller of this external function. This exists as a library intended to be usable outside of the mixer framework. You must ensure that numWorkers is not set to 0 in this codepath.

Reply:
Fair enough. Al thought the same can be said about BuildBundles() which is exported in the build package.
Ill add the check and reupload.

Reply:
I wouldn't use the `build` subpackage as a good example of a package that follows good API practices. I think it is more intended as an internal package than an external one 😕 

The swupd subpackage on the other hand should try to follow good practices for a useful API. I'm sure it **never** violates that principle. 😜

Reply:
haha I agree with you. I'm just explaining my reasoning behind the missing check. Ideally we would have checks for all those cases.